components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -373,6 +382,57 @@ public static AuthenticationContext getContextData(HttpServletRequest request) {
         return context;
     }
 
+    /**
+     * Retrieve authentication context data. If the AUTHENTICATION_CONTEXT_EXPIRY_VALIDATION is enabled and the cache
+     * entry is expired, an exception is thrown.
+     * If the AUTHENTICATION_CONTEXT_EXPIRY_VALIDATION is disabled, it will not validate the authentication context
+     * cache for session expiry.
+     *
+     * @param request Authentication request.
+     * @return Authentication context.
+     * @throws FrameworkClientException Authentication session has expired.
+     */
+    public static AuthenticationContext getContextDataWithExpiry(HttpServletRequest request)
```

Comment:
Why does this method throw a FrameworkClientException?
Shouldn't we just return null?

Reply:
That's to differentiate the expired auth context scenario. There can be different reasons for the auth context to be null.

Reply:
but the auth context can be null if it is expired right?